Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -457,9 +457,14 @@ public class GraphTraverser: GraphTraversing {
     }
 
     public func allSwiftMacroTargets(path: AbsolutePath, name: String) -> Set<GraphTarget> {
-        let dependencies = allTargetDependencies(path: path, name: name)
+        var dependencies = allTargetDependencies(path: path, name: name)
             .filter { [.staticFramework, .framework, .dynamicLibrary, .staticLibrary].contains($0.target.product) }
             .filter { self.directSwiftMacroExecutables(path: $0.path, name: $0.target.name).count != 0 }
+
+        if let target = target(path: path, name: name), !directSwiftMacroExecutables(path: path, name: name).isEmpty {
```

Comment:
Which scenario is covered by this one? I'd suggest adding a comment explaining the scenario for the next person coming across this change.